File: app/types/notifications.ts
import { User } from "@/types/index"

export type NotificationTypes = "added" | "timed" | "removed"

export type Notification = {
    id: string
    user: User
    type: NotificationTypes
    token: string
}

export type CreateNotificationRequest = {
    user: User
    type: NotificationTypes
    token: string
}

export type CreateNotificationResponse = Notification

export type GetUserNotificationsResponse = Notification[]

export type PushUserNotificationByTypeRequest = {
    env?: string | null
}

export type PushUserNotificationByTypeResponse = {
    message: string
    type: NotificationTypes
    user: User
}

export type DeleteUserNotificationResponse = {
    message: string
    type: NotificationTypes
    user: User
}

export type NotificationSettings = {
    added: boolean
    timed: boolean
    removed: boolean
    expoToken?: string | null
}

File: services/shared/contracts/notifications.go
package contracts

import "shopping-list/shared/models"

type CreateNotificationRequest struct {
	User  string                  `json:"user" validate:"required"`
	Type  models.NotificationType `json:"type" validate:"required"`
	Token string                  `json:"token" validate:"required"`
}

type CreateNotificationResponse models.Notification

type GetAllNotificationsResponse []models.Notification

type GetUserNotificationsResponse []models.Notification

type PushUserNotificationByTypeRequest struct {
	Env  string `json:"env,omitempty"`
	Text string `json:"text,omitempty"`
}
type PushUserNotificationByTypeResponse struct {
	Message string                  `json:"message"`
	Type    models.NotificationType `json:"type,omitempty"`
	User    string                  `json:"user,omitempty"`
}

type DeleteUserNotificationResponse struct {
	Message string                  `json:"message"`
	Type    models.NotificationType `json:"type,omitempty"`
	User    string                  `json:"user,omitempty"`
}
